src/components/Link/Link.tsx

```diff
@@ -22,29 +25,54 @@ export interface BaseProps {
 }
 
 export interface Props extends BaseProps {}
+export type CombinedProps = Props & WithAppProviderProps;
 
-export default function Link({
+function Link({
   url,
   children,
   onClick,
   external,
   id,
   monochrome,
-}: Props) {
+  polaris,
+}: CombinedProps) {
   const className = classNames(styles.Link, monochrome && styles.monochrome);
+  let childrenMarkup = children;
+
+  if (external && typeof children === 'string') {
+    const lastWord = children.slice(children.lastIndexOf(' ') + 1);
+    const rest = children.slice(0, children.lastIndexOf(' ') + 1);
+    const iconLabel = polaris.intl.translate(
+      'Polaris.Common.newWindowAccessibilityHint',
+    );
+
+    childrenMarkup = (
+      <React.Fragment>
+        {rest}
+        <span className={styles.IconLockup}>
+          {lastWord}
+          <Icon accessibilityLabel={iconLabel} source={ExternalSmallMinor} />
+        </span>
```

Comment:
In order to center the icon with the text, and to prevent the icon from wrapping by itself to the next line, the best solution I found is to wrap the last word and the icon in an inline-flex container.

This way of preventing unwanted wrapping won’t work with languages like Japanese because [those languages don’t use spaces between words](https://en.wikipedia.org/wiki/Japanese_writing_system#Spacing_and_punctuation). I’m not sure what to do about this.

Reply:
For languages that does not use space between words, you can grab the last character + punctuation as a word. Since each character is consider a word, a single character on the second line is perfectly fine. But punctuation should never be wrap onto the next line by itself.

✅ 
你好
嗎。

❎ 
你好嗎
。

Reply:
Thanks @michelleyschen! Would we want an icon at the end of a text link to behave the same way as punctuation?

Reply:
yes, I would consider icon a punctuation

Reply:
Thanks. I have a new implementation that doesn’t manipulate the children string. Should work for all languages now 😄